silkworm/rpc/http/connection.hpp

```diff
@@ -91,7 +92,7 @@ class Connection : public StreamWriter {
 
     static std::string get_date_time();
 
-    void compress(const std::string& clear_data, std::string& compressed_data);
+    Task<bool> compress(boost::asio::thread_pool& workers, const std::string& clear_data, std::string& compressed_data);
```

Comment:
The first parameter `workers` is unnecessary: you can use directly member variable `workers_` inside `compress` implementation